extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/graal/OpenTelemetrySdkAutoConfiguration.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.opentelemetry.runtime.graal;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+
+@TargetClass(className = "io.opentelemetry.sdk.autoconfigure.OpenTelemetrySdkAutoConfiguration")
```

Comment:
Are we calling this class at all?

I would've expected not, so I'm not sure why we need to substitute anything for GraalVM as we want the class removed, right?

Reply:
The `GlobalOpenTelemetry` class has a static method `get` that calls a method `maybeAutoConfigure` that loads the `OpenTelemetrySdkAutoConfiguration` class and invokes the method `initialize`... and in native image this makes it fail, since we have incomplete classpath and it tries to load metrics and other classes that we do not import. So I substituted it to return null so it doesn't try to autoconfigure when doing native image compilation.

Reply:
Ah, so this is only a problem because we're depending on the entire autoconfigure artifact, otherwise we'd get a CNFE and it wouldn't be called, right?

If so, can we add a comment to the substitution that it can be removed when autoconfigure SPI is a separate artifact?

Reply:
Yup, sure, just added it =]

Reply:
Yup, this just happens because we are depending on the entire artifact